TextBox.py

```diff
@@ -123,61 +143,86 @@ def replace_bytes(match: re.Match) -> str:
 
             # If this word is a line/box break, trim our line back a word and deal with it later.
             break_char = False
-            if words[end_index-1][0].code in [0x01, 0x04]:
+            if words[end_index-1][0].code in [[0x0A, 0x81A5],[0x01, 0x04]][lang]:
                 line = words[start_index:end_index-1]
                 break_char = True
 
             # Check the width of the line after adding one more word.
-            if end_index == len(words) or break_char or calculate_width(words[start_index:end_index+1]) > line_width:
+            if end_index == len(words) or break_char or (calculate_width(words[start_index:end_index+1],lang) >= line_width):
                 if line or lines:
                     lines.append(line)
                 start_index = end_index
 
             # If we've reached the end of the box, finalize it.
-            if end_index == len(words) or words[end_index-1][0].code == 0x04 or len(lines) == LINES_PER_BOX:
+            if end_index == len(words) or words[end_index-1][0].code == [0x81A5,0x04][lang] or len(lines) == line_box:
                 # Append the same icon to any wrapped boxes.
                 if icon_code and box_count > 1:
                     lines[0][0] = [icon_code] + lines[0][0]
                 processed_boxes.append(lines)
                 lines = []
                 box_count += 1
-
     # Construct our final string.
     # This is a hideous level of list comprehension. Sorry.
-    return '\x04'.join(['\x01'.join([' '.join([''.join([code.get_string() for code in word]) for word in line]) for line in box]) for box in processed_boxes])
+    if lang: return '\x04'.join(['\x01'.join([' '.join([''.join([code.get_string() for code in word]) for word in line]) for line in box]) for box in processed_boxes])
+    else: return '^'.join(['&'.join([''.join([''.join([code.get_string() for code in word]) for word in line]) for line in box]) for box in processed_boxes]).replace("&&","&").replace("^^","^").replace("&^","^")
 
 
-def calculate_width(words: list[list[TextCode]]):
+def calculate_width(words: list[list[TextCode]], lang: str|int):
     words_width = 0
+    lang= 1 if lang in ["en",1] else 0
+    CC=Messages.CONTROL_CODES if lang else Messages.CC_PARSE_JP
```

Comment:
More issues with equal sign spacing, and I think having lang be a string or an int is very awkward. Adds more reason for my previous comment to not overwrite the `lang` variable, and just always pass it as a string.